packages/graphql/src/services/sessions.ts

```diff
@@ -383,20 +446,93 @@ export async function deactivateSessionBlock(
       blocks: {
         update: {
           where: {
-            id: sessionBlockId,
+            id: Number(sessionBlockId),
           },
           data: {
             status: SessionBlockStatus.EXECUTED,
             instances: {
-              update: [],
+              update: Object.entries(instanceResults).map(([id, results]) => ({
+                where: { id: Number(id) },
+                data: {
+                  results: results.results,
+                  // TODO: persist responses or "too much information"? delete when session is completed? what about anonymous users?
+                  responses: {
```

Comment:
not sure whether it is any use to persist each response for each participant, or too much data/information. could delete it once the session is completed, or a week after, or so.